src/test/compile-fail/unreachable-try-pattern.rs

```diff
@@ -39,6 +40,7 @@ fn qux(x: Result<u32, Void>) -> Result<u32, i32> {
 fn vom(x: Result<u32, Void>) -> Result<u32, i32> {
     let y = (match x { Ok(n) => Ok(n), Err(e) => Err(e) })?;
     //~^ WARN unreachable pattern
+    //~| WARN unreachable expression
```

Comment:
Similarly here, it feels unfortunate to start warning unless we have a nicer way to write this code (but, at present, we don't...?).

If we had the `!` pattern concept, we could perhaps do 

```rust
match x { 
    Ok(n) => Ok(n),
    Err(!)
}
```